src/Storage/Storage.Management/StorageAccount/RestoreAzStorageBlobRange.cs

```diff
@@ -119,47 +122,95 @@ public override void ExecuteCmdlet()
             if (ShouldProcess(this.StorageAccountName, "Restore Blob Range"))
             {
                 if (waitForComplete)
-                {                    
-                    Task<AzureOperationResponse<BlobRestoreStatus>> beginTask = this.StorageClient.StorageAccounts.BeginRestoreBlobRangesWithHttpMessagesAsync(
-                     this.ResourceGroupName,
-                     this.StorageAccountName,
-                     this.TimeToRestore,
-                     PSBlobRestoreRange.ParseBlobRestoreRanges(this.BlobRestoreRange));
-
-                    beginTask.Wait();
-
-                    AzureOperationResponse<BlobRestoreStatus> response = beginTask.Result;
-
-                    WriteWarning(string.Format("Restore blob ranges with Id '{0}' started. Restore blob ranges time to complete is dependent on the size of the restore.", response.Body is null ? "" : response.Body.RestoreId));
+                {
+                    Track2.StorageAccountResource account = this.StorageClientTrack2.GetStorageAccount(this.ResourceGroupName, this.StorageAccountName);
+                    var restoreLro = account.RestoreBlobRanges(
+                        WaitUntil.Started,
+                        new Track2Models.BlobRestoreContent(
+                            this.TimeToRestore,
+                            PSBlobRestoreRange.ParseBlobRestoreRanges(this.BlobRestoreRange))
+                        );
+
+                    Dictionary<string, object> temp = restoreLro.GetRawResponse().Content.ToObjectFromJson() as Dictionary<string, object>;
+                    object restoreId;
+
+                    if (temp != null && temp.TryGetValue("restoreId", out restoreId))
```

Comment:
We may return early when this condition is hit, otherwise all the places using `temp.***` need a null-check.

Reply:
Discussed offline with Wei and changed to throw an exception when temp is null. Thus temp referenced below is definitely not null. 
Getting the raw response is a temporary workaround for this [SDK issue](https://github.com/Azure/azure-sdk-for-net/issues/29060). We will remove this chunk of code once the issue is fixed, and we should be able to get BlobRestoreStatus object directly with the SDK.